src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -182,90 +188,47 @@ protected virtual void ProcessTypes(AssemblyDefinition assembly, XPathNavigator
                         continue;
                 }
 
-                TypeDefinition type = assembly.MainModule.GetType(fullname);
-
-                if (type == null && assembly.MainModule.HasExportedTypes)
-                {
-                    foreach (var exported in assembly.MainModule.ExportedTypes)
-                    {
-                        if (fullname == exported.FullName)
-                        {
-                            var resolvedExternal = ProcessExportedType(exported, assembly, typeNav);
-                            if (resolvedExternal != null)
-                            {
-                                type = resolvedExternal;
-                                break;
-                            }
-                        }
-                    }
-                }
+                TypeDesc type = CustomAttributeTypeNameParser.GetTypeByCustomAttributeTypeName(assembly, fullname, throwIfNotFound: false);
 
                 if (type == null)
                 {
-                    if (warnOnUnresolvedTypes)
-                        LogWarning(typeNav, DiagnosticId.XmlCouldNotResolveType, fullname);
+                    //if (warnOnUnresolvedTypes)
+                    //    LogWarning(typeNav, DiagnosticId.XmlCouldNotResolveType, fullname);
                     continue;
                 }
 
                 ProcessType(type, typeNav);
             }
         }
 
-        protected virtual TypeDefinition? ProcessExportedType(ExportedType exported, AssemblyDefinition assembly, XPathNavigator nav) => exported.Resolve();
-
-        void MatchType(TypeDefinition type, Regex regex, XPathNavigator nav)
+        void MatchType(TypeDesc type, Regex regex, XPathNavigator nav)
         {
-            if (regex.Match(type.FullName).Success)
+            if (regex.Match(type.GetDisplayName()).Success)
                 ProcessType(type, nav);
-
-            if (!type.HasNestedTypes)
-                return;
-
-            foreach (var nt in type.NestedTypes)
-                MatchType(nt, regex, nav);
         }
 
-        protected virtual bool ProcessTypePattern(string fullname, AssemblyDefinition assembly, XPathNavigator nav)
+        protected virtual bool ProcessTypePattern(string fullname, ModuleDesc assembly, XPathNavigator nav)
         {
             Regex regex = new Regex(fullname.Replace(".", @"\.").Replace("*", "(.*)"));
 
-            foreach (TypeDefinition type in assembly.MainModule.Types)
-            {
+            foreach (TypeDesc type in assembly.GetAllTypes())
                 MatchType(type, regex, nav);
-            }
-
-            if (assembly.MainModule.HasExportedTypes)
```

Comment:
Can you add a TODO that we need to look at exported types as well? GetAllTypes won't include exported types.